### ISSUE TITLE
Regression with overloads and `: *tuple` in args

### ISSUE THREAD

**Issue by minmax** [microsoft/pyright#9238]
After update from `pyright` `1.1.384` to `1.1.385`, some types was broke.

```python
from collections.abc import Callable
from typing import Any
from typing import overload

from typing_extensions import TypeVar


AT = TypeVar("AT", infer_variance=True)
RT = TypeVar("RT", infer_variance=True)

T1 = TypeVar("T1", infer_variance=True)
T2 = TypeVar("T2", infer_variance=True)
T3 = TypeVar("T3", infer_variance=True)
T4 = TypeVar("T4", infer_variance=True)


@overload
def comp(*functions: *tuple[Callable[[AT], RT]]) -> Callable[[AT], RT]: ...
@overload
def comp(
    *functions: *tuple[Callable[[AT], T1], Callable[[T1], RT]],
) -> Callable[[AT], RT]: ...
@overload
def comp(
    *functions: *tuple[Callable[[AT], T1], Callable[[T1], T2], Callable[[T2], RT]],
) -> Callable[[AT], RT]: ...


# error: Overloaded implementation is not consistent with signature of overload 3
#    Type "(*functions: *tuple[(AT@comp) -> Any, *tuple[(Any) -> Any, ...]]) -> ((AT@comp) -> Any)" is not assignable to type "(*functions: *tuple[(AT@comp) -> T1@comp, (T1@comp) -> T2@comp, (T2@comp) -> RT@comp]) -> ((AT@comp) -> RT@comp)"
#      Missing keyword parameter "functions[2]" (reportInconsistentOverload)

def comp(
    *functions: *tuple[
        Callable[[AT], Any],
        *tuple[Callable[[Any], Any], ...],
    ],
) -> Callable[[AT], Any]: ...

```

**Comment by minmax** [comment]
Similar error, but in slightly different code.

```python
from collections.abc import Callable
from functools import partial
from typing import Concatenate
from typing import ParamSpec
from typing import overload

from typing_extensions import TypeVar


P = ParamSpec("P")
R = TypeVar("R", infer_variance=True)
T = TypeVar("T", infer_variance=True)
T1 = TypeVar("T1", infer_variance=True)
T2 = TypeVar("T2", infer_variance=True)

# 27:5 - error: Overloaded implementation is not consistent with signature of overload 2
#    Type "(func: (...) -> R@concat, /, *args: object) -> ((...) -> R@concat)" is not assignable to type "(func: (T@concat, **P@concat) -> R@concat, /, *args: *tuple[T@concat]) -> ((**P@concat) -> R@concat)"
#    Missing keyword parameter "args[0]" (reportInconsistentOverload)

@overload
def concat(func: Callable[P, R], /, *args: *tuple[()]) -> Callable[P, R]: ...
@overload
def concat(func: Callable[Concatenate[T, P], R], /, *args: *tuple[T]) -> Callable[P, R]: ...
@overload
def concat(func: Callable[Concatenate[T, T1, P], R], /, *args: *tuple[T, T1]) -> Callable[P, R]: ...
@overload
def concat(func: Callable[Concatenate[T, T1, T2, P], R], /, *args: *tuple[T, T1, T2]) -> Callable[P, R]: ...


def concat(func: Callable[..., R], /, *args: object) -> Callable[..., R]:
    return partial(func, *args)
```

**Comment by erictraut** [comment]
For my own benefit, here's a simplified example that demonstrates the issue:

```python
from typing import overload

@overload
def func(*args: *tuple[int]) -> int: ...
@overload
def func(*args: *tuple[int, int, int]) -> int: ...

def func(*args: *tuple[int, *tuple[int, ...]]) -> int: ...
```

**Comment by erictraut** [comment]
Thanks for the bug report, and apologies for the regression. This will be addressed in the next release.

If your code base is in a public github repo, it would be helpful if we could add it to the [projects list](https://github.com/hauntsaninja/mypy_primer/blob/master/mypy_primer/projects.py) for mypy_primer. This is a tool that runs as part of pyright's CI (as well as mypy and typeshed) to catch type checker regressions. If that's something you're interested in, please file a request or a PR with the mypy_primer project.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.386
